src/main/java/com/aws/greengrass/certificatemanager/certificate/CertificateGenerator.java

```diff
@@ -59,9 +60,37 @@ public abstract void generateCertificate(Supplier<List<String>> connectivityInfo
      * @return true if certificate should be regenerated, else false
      */
     public boolean shouldRegenerate() {
-        Instant dayFromNow = Instant.now(clock).plus(1, ChronoUnit.DAYS);
-        Instant expiryTime = getExpiryTime();
-        return expiryTime.isBefore(dayFromNow);
+        return isExpired() || isAboutToExpire();
+    }
+
+    /**
+     * Check if the certificate is expired.
+     *
+     * @return true if the certificate is expired
+     */
+    public boolean isExpired() {
+        return getExpiryTime().isBefore(Instant.now(clock));
+    }
+
+    /**
+     * Check if the certificate is close to expiring.
+     *
+     * @return true if the certificate is to its expiration
+     */
+    public boolean isAboutToExpire() {
+        return !isExpired() && getExpiryTime().isBefore(Instant.now(clock).plus(1, ChronoUnit.DAYS));
+    }
+
+    /**
+     * Calculate the duration that the certificate is valid for.
+     *
+     * <p>If certificate expiry is in the past, zero is returned.
+     *
+     * @return duration from current time to certificate expiry time
+     */
+    public Duration getValidity() {
```

Comment:
Similarly, this can live inside the expiry monitor as well since `getExpiryTime` is already public